_pages/docs/testing-with-xquery.md

```diff
@@ -5,17 +5,233 @@ lead_text: ''
+} catch($err) {
+ $err
+}
+```
+### Scenario 9 - Test assert equal JSON - same json node and values as json document
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $actual := json:object()
+let $_ := map:put($actual,"a",111)
+let $_ := map:put($actual,"b",222)
+
+let $actualJson := xdmp:to-json($actual)
+let $expected := json:object()
+let $_ := map:put($expected,"a",111)
+let $_ := map:put($expected,"b",222)
+let $expectedJson := xdmp:to-json($expected)
+
+return
+try{
+	test:assert-equal-json($actualJson, $expectedJson)
+} catch($err) {
+ $err
+}
+```
+### Scenario 10 - Test assert equal JSON - Node mismatch as json document and validate expected error message
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $actual := json:object()
+let $_ := map:put($actual,"a",111)
+let $_ := map:put($actual,"b",222)
+let $_ := map:put($actual,"c",333)
+
+let $actualJson := xdmp:to-json($actual)
+let $expected := json:object()
+let $_ := map:put($expected,"a",111)
+let $_ := map:put($expected,"b",222)
+let $expectedJson := xdmp:to-json($expected)
+let $errorMessageExpected := "Assert Equal Json failed (ASSERT-EQUAL-JSON-FAILED): " || $actualJson || " " || $expectedJson
+return
+try{
+	test:assert-equal-json($actualJson, $expectedJson)
+} catch($err) {
+ test:assert-equal($err/error:format-string/string(), $errorMessageExpected)
+}
+```
+### Scenario 11 - Test assert equal JSON - Node value mismatch as json document and validate expected error message
+```
+xquery version "1.0-ml";
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $actual := json:object()
+let $_ := map:put($actual,"a",111)
+let $_ := map:put($actual,"b",222)
+let $_ := map:put($actual,"c",333)
+
+let $actualJson := xdmp:to-json($actual)
+let $expected := json:object()
+let $_ := map:put($expected,"a",111)
+let $_ := map:put($expected,"b",222)
+let $_ := map:put($expected,"c",444)
+let $expectedJson := xdmp:to-json($expected)
+let $errorMessageExpected := "Assert Equal Json failed (ASSERT-EQUAL-JSON-FAILED): " || $actualJson || " " || $expectedJson
+return
+try{
+test:assert-equal-json($actualJson, $expectedJson)
+} catch($err) {
+ test:assert-equal($err/error:format-string/string(), $errorMessageExpected)
+}
+```
+### Scenario 12 - Test assert http get status - Look for a document that does n't exist after a valid authentication with correct user name and passwd
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $url := "http://localhost:8000/v1/documents?uri=document_does_not_exist.xml"
+let $options :=      
+<options xmlns="xdmp:http">
+   <authentication method="digest">
+     <username>admin</username>
+     <password>admin</password>
```

Comment:
Since this doesn't have to work, it's only an example let's change these credentials to 
```
<username>username</username>
<password>password</password>
```

admin:admin is a poor security decision, we shouldn't reference it in our examples if we can avoid it.

Reply:
Even better: marklogic-unit-test is intended to be deployed with ml-gradle, so use the substitutions that are available:

```
(: Have ml-gradle substitute the credentials :)
let $options :=      
<options xmlns="xdmp:http">
   <authentication method="digest">
     <username>%%mlUsername%%</username>
     <password>%%mlPassword%%</password>
   </authentication>
</options>
```